astropy/utils/tests/test_data.py

```diff
@@ -539,7 +1408,7 @@ def read(self, length=None):
 
     monkeypatch.setattr(urllib.request, "urlopen", mockurl)
 
-    with pytest.raises(urllib.error.ContentTooShortError):
+    with pytest.raises(urllib.error.URLError):
```

Comment:
Hmm... Why the exception type change?

Reply:
Good question. Maybe it got changed in the abortive attempt to use urlretrieve? I'll switch it back and see if I get bizarre CI failures on Windows or something.

Reply:
Ah right. This is tricky because when we consider multiple sources we have to capture all the exceptions and raise some generic URLError that covers all of them. I added a check so that if there's only one source (a common case, though not for astropy data) you get the more specific error. Not totally sure you get the completely right traceback.

Reply:
No worries. We can always revisit in the future if there is a need.